super_gui/src/components/FrontDoorUI.tsx

```diff
@@ -0,0 +1,514 @@
+import { useState, useEffect } from "react";
+import { toast } from "react-toastify";
+import Lock from "/assets/icons/bx-lock-alt.svg";
+import OpenLock from "/assets/icons/bx-lock-open-alt.svg";
+import NoSignal from "/assets/icons/bx-no-signal.svg";
+import Placeholder from "/assets/images/placeholder.jpg";
+import Package from "/assets/icons/bxs-package.svg";
+import FrontDoorModal from "./FrontDoorModal";
+
+const FrontDoorUI = () => {
+  const [lockStatus, setLockStatus] = useState<string | null>(null);
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [hasConfirmed, setHasConfirmed] = useState<boolean>(false);
+  const [guestCode, setGuestCode] = useState<string | null>("");
+  const [displayGuestCode, setDisplayGuestCode] = useState<boolean>(false);
+  const [copied, setCopied] = useState<string | null>("");
+  const [packageStatus, setPackageStatus] = useState<boolean>(true);
+
+  // Temporary until endpoints are created.
+  const togglePackageStatus = () => {
+    setPackageStatus((prev) => !prev);
+    toast.success("Package retrieved by tenant.");
+    console.log("Package retrieved by tenant.");
+  };
+
+  const doorId = 1; // Initial Door value:
+  const userId = 1; // Initial Tenant value:
+
+  // Gives prompt copy functionality
+  const handleCopy = () => {
+    setCopied(guestCode.code);
+    navigator.clipboard.writeText(guestCode.code);
+    setTimeout(() => setCopied(""), 3000);
+  };
+
+  // GET Door status
```

Comment:
for next time, I would make a service which does all the calls to the endpoints. THis way it is clear where to look for the api calls.